docs/services/ml/upload-training-script.md

```diff
@@ -10,48 +10,231 @@ no_service: true
 ---
 
 You can create your own custom Python training script that trains ML models to your specifications using the Machine Learning framework of your choice (PyTorch, Tensorflow, TFLite, ONNX, or any other framework).
-Once added to the [Viam registry](https://app.viam.com/registry?type=Training+Script), you can use the training script to build models for your datasets.
+Once added to the [Viam registry](https://app.viam.com/registry?type=Training+Script), you can use the training script to build ML models based on your datasets.
 
 {{< alert title="Note" color="note" >}}
-For a lower-code approach, you can [train TFLite classification and object detection models](/services/ml/train-model/) on the [Viam app **DATA** page](https://app.viam.com).
+For a no-code approach, you can [train TFLite classification and object detection models](/services/ml/train-model/) on the [Viam app **DATA** page](https://app.viam.com) or use existing [training scripts](https://app.viam.com/registry?type=Training+Script) from the Viam Registry.
 {{< /alert >}}
 
 Follow this guide to create, upload, and submit a Python script that loads a training dataset, trains an ML model, and produces a model artifact.
 
 ## Create a training script
 
-To start, create an entrypoint file called <file>training.py</file> in a folder called <file>model</file>.
-This file will contain the main logic of your training script.
-
-Add the following logic to your code in <file>training.py</file>:
-
-{{%expand "Step 1: Parse command line inputs" %}}
+1. Create a folder for the training-script, for example <file>my-training</file>.
+1. Inside the top level folder (in this example <file>my-training</file>), create a file called `setup.py` with the following contents:
+
+   ```python {class="line-numbers linkable-line-numbers" data-line="11"}
+   from setuptools import find_packages, setup
+
+   setup(
+       name="my-training",
```

Comment:
super nit, but this should probably be called model

Reply:
This is the convention we use for all the example training scripts but it's up to you

Reply:
It does describe the training script though right?but also it doesn't affect anything anyway, right?